Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -897,7 +911,9 @@ AXTextMarker AXTextMarker::findMarker(AXDirection direction, AXTextUnit textUnit
             currentObject = findObjectWithRuns(*currentObject, direction, stopAtID);
             if (currentObject) {
                 // We should return when the containing block is different (indicating a paragraph), or when we hit the border of an editable object.
-                if (currentRuns->containingBlock != currentObject->textRuns()->containingBlock || lastObjectIsEditable != !!currentObject->editableAncestor() || (currentObject->roleValue() == AccessibilityRole::LineBreak && !currentObject->editableAncestor()))
+                // For sentences, don't stop at line breaks, since the text break iterator needs to find the next sentence boundary, which isn't necessarily at a break.
+                bool shouldStopAtLineBreaks = textUnit == AXTextUnit::Word && currentObject->roleValue() == AccessibilityRole::LineBreak && !currentObject->editableAncestor();
+                if (currentRuns->containingBlock != currentObject->textRuns()->containingBlock || lastObjectIsEditable != !!currentObject->editableAncestor() || shouldStopAtLineBreaks)
```

Comment:
Should we put the `shouldStopAtLineBreaks` check before the `lastObjectIsEditable != !!currentObject->editableAncestor()` check since it's quite a bit cheaper?

Reply:
Yeah good call, will move that up front!